test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -0,0 +1,959 @@
+            os.remove(self.job_out1_tempfile)
+            os.remove(self.job_out2_tempfile)
+            os.remove(self.job_out3_tempfile)
+
+        except OSError:
+            pass
+
+    def read_env(self, outputfile, ishook):
+        """
+        Parse the output file and store the
+        variable list in a dictionary
+        """
+
+        with open(outputfile) as fd:
+            pkey = ""
+            ppenv = {}
+            tmp = {}
+            tmp_exclude = {}
+            for line in fd:
+                l = line.split("=", 1)
+                if (len(l) == 2):
+                    pkey = l[0]
+                    if pkey not in self.exclude_env:
+                        tmp[pkey] = l[1]
+                        ppenv = tmp
+                    else:
+                        tmp_exclude[pkey] = l[1]
+                        ppenv = tmp_exclude
+                elif pkey != "":
+                    # append to previous dictionary entry
+                    ppenv[pkey] += l[0]
+        if (ishook == "hook"):
+            self.env_hook = tmp
+            self.env_hook_exclude = tmp_exclude
+        else:
+            self.env_nohook = tmp
+            self.env_nohook_exclude = tmp_exclude
+
+    def common_log_match(self, daemon):
+        """
+        Validate the env variable output in daemon logs
+        """
+        logmsg = ["TEST_COMMA=1\,2\,3\,4",
+                  "TEST_SEMICOLON=;",
+                  "TEST_ENCLOSED=\\'\,\\'",
+                  "TEST_COLON=:",
+                  "TEST_BACKSLASH=\\\\",
+                  "TEST_DQUOTE=\\\"",
+                  "TEST_DQUOTE2=happy days\\\"are\\\"here to stay",
+                  "TEST_DQUOTE3=nothing compares\\\" to you",
+                  "TEST_DQUOTE4=\\\"music makes the people\\\"",
+                  "TEST_DQUOTE5=music \\\"makes \\'the\\'\\\"people",
+                  "TEST_DQUOTE6=lalaland\\\"",
+                  "TEST_SQUOTE=\\'",
+                  "TEST_SQUOTE2=happy\\'days",
+                  "TEST_SQUOTE3=the days\\'are here now\\'then",
+                  "TEST_SQUOTE4=\\'the way that was\\'",
+                  "TEST_SQUOTE5=music \\'makes \\\"the\\'\\\"people",
+                  "TEST_SQUOTE6=loving\\'",
+                  "TEST_SPECIAL={}[]()~@#$%^&*!",
+                  "TEST_SPECIAL2=<dumb-test_text>"]
+
+        if (daemon == "mom"):
+            self.logger.info("Matching in mom logs")
+        elif (daemon == "server"):
+            self.logger.info("Matching in server logs")
+        else:
+            self.logger.info("Provide a valid daemon name; server or mom")
+            return
+
+        for msg in logmsg:
+            if (daemon == "mom"):
+                # Match the trailing separator (',')
+                self.mom.log_match(msg + ',', max_attempts=3)
+            elif (daemon == "server"):
+                self.server.log_match(msg, starttime=self.server.ctime)
+
+        # Following lines are commented due to PTL bug PP-1008
```

Comment:
remove the commented lines of code

Reply:
We're supposed to keep the commented lines, which would get uncommented once PP-1008 is fixed.